tests/end2end/playwright/edition-form.spec.js

```diff
@@ -510,3 +510,262 @@ test.describe('Multiple geometry layers', () => {
+        await expect(firstTabFields.nth(4)).toHaveText('Widget_test');
+
+        // second popup tab
+        const secondTabFields = page.locator('.lizmapPopupContent > .lizmapPopupSingleFeature .lizmapPopupDiv div.tab-pane').nth(1).locator('.field');
+        
+        await expect(secondTabFields).toHaveCount(1);
+        await expect(secondTabFields.nth(0)).toHaveText('[%kk[% "ids" %]%]');
+
+        // third popup tab
+        const thirdTabFields = page.locator('.lizmapPopupContent > .lizmapPopupSingleFeature .lizmapPopupDiv div.tab-pane').nth(2).locator('.field');
+
+        await expect(thirdTabFields).toHaveCount(1);
+        await expect(thirdTabFields.nth(0)).toHaveText('Widget_test');
+
+        // edit the feature
+        let editFeatureRequestPromise = page.waitForResponse(response => response.url().includes('editFeature'));
+        await page.locator('.feature-toolbar > button:nth-child(5)').click();
+        await editFeatureRequestPromise;
+
+        // the text widgets in the form should be filled as in the popup
+        const form = page.locator('#jforms_view_edition');
+        await expect(form.locator('div.tab-pane')).toHaveCount(3);
+
+        // first tab
+        let firstFormTabFields = await form.locator('div.tab-pane').nth(0).locator('div.control-group');
+        await expect(firstFormTabFields).toHaveCount(5);
+
+        await expect(firstFormTabFields.nth(0).locator('label')).toHaveText('id');
+        await expect(firstFormTabFields.nth(0).locator('input')).toHaveValue('1');
+
+        await expect(firstFormTabFields.nth(1).locator('label')).toHaveText('Constant');
+        let trimText = (await firstFormTabFields.nth(1).locator('input').inputValue()).replace(/\s+/g,' ').trim()
+        expect(trimText).toBe('text string')
+
+        await expect(firstFormTabFields.nth(2).locator('label')).toHaveText('Geometry X');
+        trimText = (await firstFormTabFields.nth(2).locator('input').inputValue()).replace(/\s+/g,' ').trim();
+        expect(trimText).toBe('3.860212606866182');
+
+        await expect(firstFormTabFields.nth(3).locator('label')).toHaveText('id check');
+        trimText = (await firstFormTabFields.nth(3).locator('input').inputValue()).replace(/\s+/g,' ').trim();
+        expect(trimText).toBe('1');
+
+        await expect(firstFormTabFields.nth(4).locator('label')).toHaveText('Name');
+        await expect(firstFormTabFields.nth(4).locator('input')).toHaveValue('Widget_test');
+
+        // second tab
+        let secondFormTabFields = await form.locator('div.tab-pane').nth(1).locator('div.control-group');
+        await expect(secondFormTabFields).toHaveCount(1);
+
+        await expect(secondFormTabFields.nth(0).locator('label')).toHaveText('Wrong expression');
+        trimText = (await secondFormTabFields.nth(0).locator('input').inputValue()).replace(/\s+/g,' ').trim();
+        // wrong expressions should not be evaluated
+        expect(trimText).toBe('[%kk[% "ids" %]%]');
```

Comment:
Not sure if this is the best for the end-user ?
Maybe an translated text "An error occurred, please contact the GIS technician" ?
Any idea ?
or depending if we are connected as admin or not ?

Reply:
Do you mean to return a message instead of the "raw" expression if this same expression is not evaluated correctly?

Reply:
Yes, I don't think showing QGIS expressions on the webbrowser side is a good thing.
I don't think so, but the end-user (not the GIS technician) doesn't know what is this. I would prefer an error message ? (at least for non admin users)

Maybe expressions contain sensitive information (low probability I think)

Reply:
Got it and I agree.

Something like that?

```
Not admin users -> "Error: expression not evaluated"
Admin user -> "Error: the following expression <> was not evaluted"
```
or simply, regardless of the connected user:

```
"Error: expression not evaluated"
```

assuming that the adim knows he will have to control the expression

Reply:
If an error occurs while evaluating the expression : 
* the log should be visible in the administration panel http://localhost:8130/admin.php/admin/logs to give some hints about something went wrong. Ideally `project XXX, error while evaluating the expression 'YYY' in the text widget"
* if the user an administrator, maybe with using https://github.com/3liz/lizmap-web-client/blob/20eee3ff528ed5af3efc9d4fd893a662cf9e199d/lizmap/modules/view/controllers/lizMap.classic.php#L574 the user should have some debug info, or tell me to consult the log page
* if the user is not an administrator,  I don't think that the word `expression` must be displayed to the user in the webinterface. In Lizmap, a lof of users doesn't what is QGIS, what is a "QGIS expression" etc. The error message must stay IMHO "generic", like "An error occurred, please contact the GIS administrator".

CC @rmartin if you can confirm ?

Reply:
Ok sorry, I just realized that the wrong expression can be in a "parent" text like : 

```
We have a
[%kk[% 'blue' %]%]
sky today.
```
Correct ?
I don't think the expression service can tell us this error (according to the QGIS API)

Reply:
> Correct ? I don't think the expression service can tell us this error (according to the QGIS API)

Yes, 
this is the main problem about error handling. Server returns the raw expression if evaluation fails.

Probably this error should be catch in the `replace_expression_text` function:

https://github.com/3liz/qgis-lizmap-server-plugin/blob/eb8bda30e0e16a044ef2688a904a61cd9cef6a37/lizmap_server/expression_service.py#L331

but it is not clear to me how. 

Since the wrong expression could be anything, it is hard to detect a pattern that defines it.

Unfortunately static method `repaceExpressionText` of `QgsExpression` class return only a string (maybe useless in this case?)

https://qgis.org/pyqgis/master/core/QgsExpression.html#qgis.core.QgsExpression.replaceExpressionText

Solving on plugin side and manage response on client side could be useful to handling errors also on Popup.

Reply:
Yes sorry, I don't think we can do something here, as even the QGIS API doesn't